fdbserver/SimulatedCluster.actor.cpp

```diff
@@ -1596,34 +1586,43 @@ void SimulationConfig::setTenantMode(const TestConfig& testConfig) {
 }
 
 void SimulationConfig::setEncryptionAtRestMode(const TestConfig& testConfig) {
-	EncryptionAtRestMode encryptionMode = EncryptionAtRestMode::DISABLED;
-	// Only Redwood support encryption. Disable encryption if non-Redwood storage engine is explicitly specified.
-	bool disableEncryption = testConfig.disableEncryption ||
-	                         (testConfig.storageEngineType.present() && testConfig.storageEngineType.get() != 3);
-	// TODO: Remove check on the ENABLE_ENCRYPTION knob once the EKP can start using the db config
-	if (!disableEncryption && (SERVER_KNOBS->ENABLE_ENCRYPTION || !testConfig.encryptModes.empty())) {
-		TenantMode tenantMode = db.tenantMode;
-		if (!testConfig.encryptModes.empty()) {
-			std::vector<EncryptionAtRestMode> validEncryptModes;
-			// Get the subset of valid encrypt modes given the tenant mode
-			for (int i = 0; i < testConfig.encryptModes.size(); i++) {
-				EncryptionAtRestMode encryptMode = EncryptionAtRestMode::fromString(testConfig.encryptModes.at(i));
-				if (encryptMode != EncryptionAtRestMode::DOMAIN_AWARE || tenantMode == TenantMode::REQUIRED) {
-					validEncryptModes.push_back(encryptMode);
-				}
-			}
-			if (validEncryptModes.size() > 0) {
-				encryptionMode = deterministicRandom()->randomChoice(validEncryptModes);
-			}
-		} else {
-			// TODO: These cases should only trigger with probability (BUGGIFY) once the server knob is removed
-			if (tenantMode == TenantMode::DISABLED || tenantMode == TenantMode::OPTIONAL_TENANT || BUGGIFY) {
-				// optional and disabled tenant modes currently only support cluster aware encryption
-				encryptionMode = EncryptionAtRestMode::CLUSTER_AWARE;
-			} else {
-				encryptionMode = EncryptionAtRestMode::DOMAIN_AWARE;
-			}
+	// Enable encryption with higher probability than disabling it in simulation. Given encryption is an add-on style
+	// feature, enabling it still provide good test coverage for the rest of the code.
+	static const double probability[EncryptionAtRestMode::END] = { 0.25, 0.5, 0.25 };
+	std::vector<int> available;
+	if (!testConfig.encryptModes.empty()) {
```

Comment:
In this case we want to make sure that one of the modes specified is chosen for sure (not with probability)

Reply:
Also in this case we want to ensure that the encrypt mode aligns with the tenant mode. For example if we specify `['cluster_aware', 'domain_aware']` and all tenant modes we should make sure to only choose `domain_aware` when the tenant mode is required

Reply:
yeah, the issue is found by correctness. updated the code since.

Reply:
@sfc-gh-nwijetunga I missed your comment regarding when encryptModes are specified, mode assignment should not come with probability (i.e. with equal probability). updated.